### DISCUSSION TITLE
Automerge not working, RenovateBot seems to find no check runs

### DISCUSSION THREAD

**Discussion by AlCalzone** [renovatebot/renovate#10697] [Request Help]
Example here:
https://github.com/zwave-js/node-zwave-js/pull/2944

The most recent bot logs contain this:
```
DEBUG: No check runs found(branch="renovate/babel-monorepo")
{
  "result": {
    "total_count": 0,
    "check_runs": []
  }
}
```
but there are check runs defined and required:
![grafik](https://user-images.githubusercontent.com/17641229/124272585-398e9400-db3f-11eb-9d7a-a6ef0d5adae5.png)

The branch protection rule includes `renovate`, so this should not be a permission issue.

**Comment by rarkins** [comment]
Those might be status checks, not check runs. Renovate checks for both.

**Reply by AlCalzone** [reply]
I've yet to see an auto-merge happen. That particular PR has been force-pushed a couple of times, each time the PR checks were green, no merge conflicts and the summary contained
> 🚦 Automerge: Enabled.

**Reply by viceice** [reply]
Renovate needs two runs for automerge.

First run it makes sure branch is up-to-date. Then it breaks, because of pending checks.

On second run it checks again if branch is up-to-date and automerge if true.

The second run will be one hour after first run, because the hosted app runs hourly by default.

**Reply by AlCalzone** [reply]
I see, so patience is key. How does Renovate handle multiple open PRs if they cause conflicts after merging? I'm guessing one auto-merge per run or something like that?

**Reply by viceice** [reply]
Correct, it does a second run after automerge to update other prs and the dependency dashboard.

**Comment by AlCalzone** [comment]
Okay, I need to come back to this...
https://github.com/zwave-js/node-zwave-js/pull/2979 was last edited 17 hours ago and has **🚦 Automerge: Enabled.**

RenovateBot has performed several runs over the repo since then:
![grafik](https://user-images.githubusercontent.com/17641229/124590231-335c2880-de5b-11eb-8934-2f5c7ea0c664.png)

And all required checks are green:
![grafik](https://user-images.githubusercontent.com/17641229/124590316-47078f00-de5b-11eb-80db-7fafa0103cd4.png)

I'm not sure what else is preventing the automerge here.

**Reply by viceice** [reply]
check the logs for the branch at https://app.renovatebot.com/dashboard#github/zwave-js/node-zwave-js

**Reply by AlCalzone** [reply]
It claims that a PR review is required:
```
DEBUG: Automerging #2979(branch="renovate/typescript-eslint-monorepo")
DEBUG: mergePr(2979, renovate/typescript-eslint-monorepo)(branch="renovate/typescript-eslint-monorepo")
DEBUG: Branch protection: Attempting to merge PR when PR reviews are enabled(branch="renovate/typescript-eslint-monorepo")
{
  "prNo": 2979
}
DEBUG: Branch protection: Cannot automerge PR until there is an approving review(branch="renovate/typescript-eslint-monorepo")
{
  "prNo": 2979
}
```

but I don't have that enabled for the target branch `next`:
![grafik](https://user-images.githubusercontent.com/17641229/124595530-6dc8c400-de61-11eb-8803-517f57dad056.png)

**Reply by rarkins** [reply]
Needs implementation of https://github.com/renovatebot/renovate/issues/7077

**Reply by AlCalzone** [reply]
Okay, now I'm confused. These PRs (among others) were merged just fine:
https://github.com/zwave-js/node-zwave-js/pull/2972
https://github.com/zwave-js/node-zwave-js/pull/2968
https://github.com/zwave-js/node-zwave-js/pull/2967

And this one just got auto-merged aswell:
https://github.com/zwave-js/node-zwave-js/pull/2984

**Comment by nicojs** [comment]
I've got a similar issue.

Automerge doesn't seem to be working for a public github repository.

See https://github.com/stryker-mutator/stryker-js/pull/3591 (i will not merge this manually for a while).

![image](https://user-images.githubusercontent.com/1828233/175563065-0eac68d9-b5f8-4ccb-8101-e046c53d5c1b.png)

In the most recent logs (2 hours old) I see this:

```
(branch="renovate/eslint-8.x")
{
  "commitSha": "43ad3583902211c27c05ad28a998633a633ad4be"
}
DEBUG: Checking if we can automerge branch(branch="renovate/eslint-8.x")
DEBUG: mergeStatus=no automerge(branch="renovate/eslint-8.x")
DEBUG: Ensuring PR(branch="renovate/eslint-8.x")
DEBUG: There are 0 errors and 0 warnings(branch="renovate/eslint-8.x")
DEBUG: getBranchPr(renovate/eslint-8.x)(branch="renovate/eslint-8.x")
DEBUG: findPr(renovate/eslint-8.x, undefined, open)(branch="renovate/eslint-8.x")
DEBUG: Found PR #3591(branch="renovate/eslint-8.x")
DEBUG: Found existing PR(branch="renovate/eslint-8.x")
DEBUG: resolveBranchStatus(branchName=renovate/eslint-8.x, ignoreTests=false)(branch="renovate/eslint-8.x")
DEBUG: getBranchStatus(renovate/eslint-8.x)(branch="renovate/eslint-8.x")
DEBUG: branch status check result(branch="renovate/eslint-8.x")
{
  "state": "pending",
  "statuses": []
}
DEBUG: No check runs found(branch="renovate/eslint-8.x")
{
  "result": {
    "total_count": 0,
    "check_runs": []
  }
}
DEBUG: Branch status yellow(branch="renovate/eslint-8.x")
DEBUG: Processing existing PR(branch="renovate/eslint-8.x")
DEBUG: Pull Request #3591 does not need updating(branch="renovate/eslint-8.x")
DEBUG: PR is configured for automerge(branch="renovate/eslint-8.x")
DEBUG: Checking schedule(at any time, null)(branch="renovate/eslint-8.x")
DEBUG: No schedule defined(branch="renovate/eslint-8.x")
DEBUG: resolveBranchStatus(branchName=renovate/eslint-8.x, ignoreTests=false)(branch="renovate/eslint-8.x")
DEBUG: getBranchStatus(renovate/eslint-8.x)(branch="renovate/eslint-8.x")
DEBUG: branch status check result(branch="renovate/eslint-8.x")
{
  "state": "pending",
  "statuses": []
}
DEBUG: No check runs found(branch="renovate/eslint-8.x")
{
  "result": {
    "total_count": 0,
    "check_runs": []
  }
}
DEBUG: Branch status yellow(branch="renovate/eslint-8.x")
DEBUG: PR is not ready for merge (branch status is yellow)(branch="renovate/eslint-8.x")
```

It seems like build status was yellow at the time, but has since been green and should now be merged. Am I missing something obvious?

Thanks in advance!

**Reply by rarkins** [reply]
Open your dependency dashboard and tick the "request renovate to run again on this repo" checkbox. You may also be interested in platformAutomerge=true

**Reply by nicojs** [reply]
Wow, that worked 😃. So just to be clear, I configured it correctly, but renovate didn't merge because it isn't automatically triggered after workflows complete?

Followup question if yes: I've enabled platformAutomerge, does that fix the issue, because PRs get configured with automerge on?

**Reply by rarkins** [reply]
You are right on both. Passing workflows is not enough to trigger the bot to run again immediately. GitHub doesn't have the concept of a "branch is ready to merge" webhook event and instead can send dozens or even hundreds of "workflow passed" events 

But platform automerge solves it because GitHub itself can then do the automerge on Renovate's behalf
